File: custom-scripts/merge-pdfs.R
# Add path for R libraries to be recognized
# Source: https://stackoverflow.com/a/65543754
.libPaths("C:/Users/HP/Documents/R/win-library/4.1")

# Load courses' paths
courses_names <- "./courses.json" |>
  jsonlite::read_json() |>
  names()

courses_paths <- paste0("./courses/", courses_names, "/apuntes/")

# Separate paths per course into a list
for (course_path in courses_paths) {
  classes_pdfs <- list.files(course_path)
  if (length(classes_pdfs) > 0) {
    classes_pdfs_paths <- paste0(course_path, classes_pdfs) 
    tryCatch({
      qpdf::pdf_combine(
        input = classes_pdfs_paths,
        output = sub(
          "apuntes/", "merged.pdf", course_path
        )
      )
    }, error = \(e) print(e)
    )
  }
}

File: custom-scripts/merge-pdfs.py
from json import load

# Load courses' paths
courses = open("./courses.json")
courses = load(courses).keys()

# Retrieve courses' notes' path
courses_notes_path = ["./courses/" + x + "/apuntes/" for x in courses]

from os import listdir
from PyPDF2 import PdfMerger

def merge_course_pdfs(course_notes_path):
  # Retrieve course notes
  notes = [path for path in listdir(course_notes_path) if path.endswith(".pdf")]
  
  merger = PdfMerger()
  [merger.append(course_notes_path + pdf) for pdf in notes]

  # Save in apuntes folder
  merged_pdf_path = course_notes_path.replace("apuntes/", "merged.pdf")
  with open(merged_pdf_path, "wb") as new_file:
    merger.write(new_file)


for course_notes_path in courses_notes_path:
  merge_course_pdfs(course_notes_path)
  print(course_notes_path)